src/Injector.js

```diff
@@ -14,94 +9,50 @@ export default class ReduxInjector extends Component {
     children: PropTypes.func.isRequired,
     actions: PropTypes.objectOf(
       PropTypes.func.isRequired
-    ).isRequired,
-    stores: PropTypes.objectOf(
-      PropTypes.func.isRequired
     ).isRequired
   }
 
   static defaultProps = {
-    stores: {},
     actions: {}
   };
 
   constructor(props, context) {
     super(props, context);
-    this.handleChange = this.handleChange.bind(this);
-    this.update(props);
-  }
 
-  componentWillReceiveProps(nextProps) {
-    this.update(nextProps);
+    this.handleChange = this.handleChange.bind(this);
+    this.unsubscribe = context.redux.subscribe(this.handleChange);
   }
 
   componentWillUnmount() {
     this.unsubscribe();
   }
 
-  update(props) {
-    const { stores, actions } = props;
-
-    if (process.env.NODE_ENV !== 'production') {
-      invariant(
-        isPlainObject(actions),
-        '"actions" must be a plain object with functions as values. Instead received: %s.',
-        actions
-      );
-      invariant(
-        isPlainObject(stores),
-        '"stores" must be a plain object with functions as values. Instead received: %s.',
-        stores
-      );
-      Object.keys(actions).forEach(key =>
-        invariant(
-          isFunction(actions[key]),
-          'Expected "%s" in "actions" to be a function. Instead received: %s.',
-          key,
-          actions[key]
-        )
-      );
-      Object.keys(stores).forEach(key =>
-        invariant(
-          isFunction(stores[key]),
-          'Expected "%s" in "stores" to be a function. Instead received: %s.',
-          key,
-          stores[key]
-        )
-      );
-    }
-
-    const { wrapActionCreator, observeStores } = this.context.redux;
-    this.actions = mapValues(props.actions, wrapActionCreator);
+  performAction(actionCreator, ...args) {
```

Comment:
You could write a custom `Injector` (should we provide `makeInjector`?) that does something fancy to action creators, like handling promises. Relevant for #45.